### ISSUE TITLE
State is "unknown" instead of total m3

### ISSUE THREAD

**Issue by gllmlbrt** [maalerportal/maalerportal#2]
Hi, 
Thank you for this great integration.
The login went great, and it downloads the statistics as expected.

However, the state of the entity created, is always reported as unknown but instead I believe it shoudl be the total m3 value of the meter.
![image](https://github.com/user-attachments/assets/75e79a19-1898-4f4c-ac5b-7f4a96cd3fb3)
![image](https://github.com/user-attachments/assets/9355ef3f-66c9-47c6-b233-8c2932d38881)

It shoudl be possible as the data is available from the public API in place.

**Comment by gllmlbrt** [comment]
Here is the log of an error which may be the cause:
```
2024-08-08 14:20:00.149 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open with args ('/root/.netrc',) inside the event loop by custom integration 'maalerportal' at custom_components/maalerportal/sensor.py, line 129: response = await self._api.api_homeassistant_partial_post([request]) (offender: /usr/local/lib/python3.12/netrc.py, line 74: with open(file, encoding="utf-8") as fp:), please create a bug report at https://github.com/maalerportal/maalerportal/issues
For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#open
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 1048, in _async_update_entity_states
    await entity.async_update_ha_state(True)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 942, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1300, in async_device_update
    await self.async_update()
  File "/config/custom_components/maalerportal/sensor.py", line 99, in async_update
    await self._get_data(lastest_statistic)
  File "/config/custom_components/maalerportal/sensor.py", line 129, in _get_data
    response = await self._api.api_homeassistant_partial_post([request])
```

**Comment by DragonFoy** [comment]
We can see this creates a new bug, we will implement a separate entity for the history.

**Comment by Haulund83** [comment]
Jeg får ej heller data ind. Tror vi det kommer til at virke, eller skal jeg pushe vandværket for en wireless mbus key?

**Comment by DragonFoy** [comment]
> > We can see this creates a new bug, we will implement a separate entity for the history.
> 
> Has there been some plan made for this ? Basically the idea of having two entities, one for statistics and another one for total-increasing value is fine. So both woudl be added to HA dashboard and both the cost and stastistics can be presented there.

We have held off for a bit, as we have been preparing to include more data (ambient temperature, water temperature, daily flow, supply prices, and more), which we can present in both our Home Assistant and Homey solutions – but first and foremost in the Målerportal app.

Once everything is ready, we will start updating the external integrations. We hope this can happen within the coming month.

If you experience any urgent issues, please don’t hesitate to contact our support team at support@meterportal.eu – just include your address and email so we can assist you more quickly.

**Comment by lauer** [comment]
@DragonFoy  So just to be sure, that even we can see the usage on the meterportal.eu, you would not provide the information through the api, before "coming month"?

**Comment by Haulund83** [comment]
Hej

Nej, der står unknown på entiteten.

2025-09-24 05:57:05.454 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to load_default_certs with args (<ssl.SSLContext object at 0x7f2c6b8ff0>, <Purpose.SERVER_AUTH: _ASN1Object(nid=129, shortname='serverAuth', longname='TLS Web Server Authentication', oid='1.3.6.1.5.5.7.3.1')>) inside the event loop by custom integration 'maalerportal' at custom_components/maalerportal/__init__.py, line 27: api_client = ApiClient(api_config) (offender: /usr/local/lib/python3.13/ssl.py, line 722: context.load_default_certs(purpose)), please create a bug report at https://github.com/maalerportal/maalerportal/issues
For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#load_default_certs
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 156, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 712, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 683, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 2050, in _run_once
    handle._run()
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 28, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 533, in config_entry_disable
    success = await hass.config_entries.async_set_disabled_by(
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2309, in async_set_disabled_by
    reload_result = await self.async_reload(entry_id)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2273, in async_reload
    await async_setup_component(self.hass, entry.domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 170, in async_setup_component
    result = await _async_setup_component(hass, domain, config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 470, in _async_setup_component
    await asyncio.gather(
  File "/usr/src/homeassistant/homeassistant/setup.py", line 472, in <genexpr>
    create_eager_task(
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 44, in create_eager_task
    return Task(coro, loop=loop, name=name, eager_start=True)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 896, in async_setup_locked
    await self.async_setup(hass, integration=integration)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 662, in async_setup
    await self.__async_setup_with_context(hass, integration)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 751, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/maalerportal/__init__.py", line 27, in async_setup_entry
    api_client = ApiClient(api_config)

2025-09-24 05:57:05.465 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to set_default_verify_paths with args (<ssl.SSLContext object at 0x7f2c6b8ff0>,) inside the event loop by custom integration 'maalerportal' at custom_components/maalerportal/__init__.py, line 27: api_client = ApiClient(api_config) (offender: /usr/local/lib/python3.13/ssl.py, line 534: self.set_default_verify_paths()), please create a bug report at https://github.com/maalerportal/maalerportal/issues
For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#set_default_verify_paths
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 156, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 712, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 683, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 2050, in _run_once
    handle._run()
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 28, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 533, in config_entry_disable
    success = await hass.config_entries.async_set_disabled_by(
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2309, in async_set_disabled_by
    reload_result = await self.async_reload(entry_id)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2273, in async_reload
    await async_setup_component(self.hass, entry.domain, self._hass_config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 170, in async_setup_component
    result = await _async_setup_component(hass, domain, config)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 470, in _async_setup_component
    await asyncio.gather(
  File "/usr/src/homeassistant/homeassistant/setup.py", line 472, in <genexpr>
    create_eager_task(
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 44, in create_eager_task
    return Task(coro, loop=loop, name=name, eager_start=True)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 896, in async_setup_locked
    await self.async_setup(hass, integration=integration)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 662, in async_setup
    await self.__async_setup_with_context(hass, integration)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 751, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/maalerportal/__init__.py", line 27, in async_setup_entry
    api_client = ApiClient(api_config)

2025-09-24 05:57:27.287 ERROR (MainThread) [homeassistant] Error doing job: Unclosed client session (None)


________________________________

**Comment by Haulund83** [comment]
Er der noget nyt?

**Comment by DragonFoy** [comment]
Please remove any old integration and set up the new updated version. If any new issues arise, feel free to create a new issue.
